### PR TITLE
FIX, Improve integration for the tuya siren

### PR DESCRIPTION
**Pull request by Smanar**
This device works actualy only for the manufacture number = 0x0000.

Not working totaly, need too the PR https://github.com/dresden-elektronik/deconz-rest-plugin/pull/4616 because of R_GetProductId issue.

### CODE REVIEW

File: de_web_plugin.cpp [hunk 1]
```diff
@@ -5059,7 +5059,8 @@ void DeRestPluginPrivate::addSensorNode(const deCONZ::Node *node, const deCONZ::
                     {
                         fpCarbonMonoxideSensor.inClusters.push_back(IAS_ZONE_CLUSTER_ID);
                     }
-                    else if (!node->nodeDescriptor().isNull() && node->nodeDescriptor().manufacturerCode() == VENDOR_NONE)
+                    else if ((!node->nodeDescriptor().isNull() && node->nodeDescriptor().manufacturerCode() == VENDOR_NONE) ||
+                               node->nodeDescriptor().manufacturerCode() == VENDOR_EMBER)
```
Comment (manup) [inline de_web_plugin.cpp:5063]: I wonder if the nodeDescriptor checks in the `else if` should be simplified to :

```cpp
else if (isTuyaManufacturerName(manufacturer))
{
}
```

Does this makes sense?

**Review by manup — COMMENTED** [review]
Added question if we need the node descriptor check here, I'm not sure :)

File: de_web_plugin.cpp (hunk 1 above)
Comment (Smanar) [reply de_web_plugin.cpp]: Yes for the moment all device are from tuya manufacture, so no problem, have modified.
In my mind was faster to compare 2 manufactures number, than use regex.

Comment (manup) [reply de_web_plugin.cpp]: Thanks, faster yes but even on the RPi I expect the code to run in less than 1ms :)
It's also only run once during setup.

**Review by manup — APPROVED** [review]

**Pull request merged**